src/common/hobject.cc

```diff
@@ -210,14 +212,18 @@ void hobject_t::generate_test_instances(list<hobject_t*>& o)
 
 static void append_out_escaped(const string &in, string *out)
 {
-  for (string::const_iterator i = in.begin(); i != in.end(); ++i) {
-    if (*i == '%' || *i == ':' || *i == '/' || *i < 32 || *i >= 127) {
-      out->push_back('%');
-      char buf[3];
-      snprintf(buf, sizeof(buf), "%02x", (int)(unsigned char)*i);
+  for (auto c : in) {
+    int i = (int)(unsigned char)(c);
+    if (i <= 0x0f) {
```

Comment:
Why `<= 0x0f` when previously it was `< 32`?

Reply:
Just what happens when one resolves a review comment too late at night.
Now it's correct: '<=0x0f' handled separately from the rest of the <32